phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterTableWithViewsIT.java

```diff
@@ -1216,5 +1218,248 @@ public void testDroppingIndexedColDropsViewIndex() throws Exception {
+            }
+            globalViewDDLTimestamp = getLastDDLTimestamp(conn,
+                globalViewFullName);
+            assertEquals(tableDDLTimestamp, globalViewDDLTimestamp);
+
+            conn.createStatement().execute(tableColumnDropDDL);
+            tableDDLTimestamp = getLastDDLTimestamp(conn, dataTableFullName);
+            try (Connection tenantConn = DriverManager.getConnection(getUrl(), props)) {
+                long tenantViewDDLTimestamp = getLastDDLTimestamp(tenantConn,
+                    tenantViewFullName);
+                assertEquals(tableDDLTimestamp, tenantViewDDLTimestamp);
+            }
+            globalViewDDLTimestamp = getLastDDLTimestamp(conn,
+                globalViewFullName);
+            assertEquals(tableDDLTimestamp, globalViewDDLTimestamp);
+        }
+
+        //now add / drop a column from the global view and make sure it doesn't change its
+        // parent (the base table) but does change the timestamp for its child (the tenant view)
+        String globalViewColumnAddDDL = "ALTER VIEW " + globalViewFullName + " ADD COL5 varchar" +
+            "(50) " + "NULL ";
+        String globalViewColumnDropDDL = "ALTER VIEW " + globalViewFullName + " DROP COLUMN COL5 ";
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.createStatement().execute(globalViewColumnAddDDL);
+            globalViewDDLTimestamp = getLastDDLTimestamp(conn, globalViewFullName);
+            long newTableDDLTimestamp = getLastDDLTimestamp(conn,
+                dataTableFullName);
+            //table DDL timestamp shouldn't have changed
+            assertEquals(tableDDLTimestamp, newTableDDLTimestamp);
+            try (Connection tenantConn = DriverManager.getConnection(getUrl(), props)) {
+                long tenantViewDDLTimestamp = getLastDDLTimestamp(tenantConn,
+                    tenantViewFullName);
+                //but tenant timestamp should have changed
+                assertEquals(globalViewDDLTimestamp, tenantViewDDLTimestamp);
+            }
+
+            conn.createStatement().execute(globalViewColumnDropDDL);
+            globalViewDDLTimestamp = getLastDDLTimestamp(conn, globalViewFullName);
+            newTableDDLTimestamp = getLastDDLTimestamp(conn,
+                dataTableFullName);
+            //table DDL timestamp shouldn't have changed
+            assertEquals(tableDDLTimestamp, newTableDDLTimestamp);
+            try (Connection tenantConn = DriverManager.getConnection(getUrl(), props)) {
+                long tenantViewDDLTimestamp = getLastDDLTimestamp(tenantConn,
+                    tenantViewFullName);
+                //but tenant timestamp should have changed
+                assertEquals(globalViewDDLTimestamp, tenantViewDDLTimestamp);
+            }
+        }
+
+    }
+
+    public static long getLastDDLTimestamp(Connection conn, String dataTableFullName) throws SQLException {
```

Comment:
Should we move this helper method to CreateTableIT since we are doing almost same thing in CreateTableIT#verifyLastDDLTimestamp and additional timestamp bounds verification.